Source/WebCore/page/Quirks.cpp

```diff
@@ -718,6 +718,27 @@ bool Quirks::shouldOmitHTMLDocumentSupportedPropertyNames()
 #endif
 }
 
+// rdar://110097836
+bool Quirks::shouldSilenceResizeObservers() const
+{
+#if PLATFORM(IOS) || PLATFORM(VISION)
+    if (!needsQuirks())
+        return false;
+
+    // ResizeObservers are silenced on YouTube during the 'homing out' snapshout sequence to
+    // resolve rdar://109837319. This is due to a bug on the site that is causing unexpected
+    // content layout and can be removed when it is addressed.
+    auto* page = m_document->page();
+    if (!page || !page->isTakingSnapshotsForApplicationSuspension())
+        return false;
+
+    auto host = m_document->topDocument().url().host();
+    return equalLettersIgnoringASCIICase(host, "youtube.com"_s) || host.endsWithIgnoringASCIICase(".youtube.com"_s);
```

Comment:
Elsewhere we do
```
    auto host = topDocument.url().host();
    auto isYouTube = host.endsWith(".youtube.com"_s) || host == "youtube.com"_s;
```
Why is this different?

Reply:
I was trying to make it similar to the one we have for the Window resize events. We seem to do a few different things including `RegistrableDomain(m_document->url()).string()`

I'll double check which option is the correct/preferred way to do this host testing

Reply:
I filed https://bugs.webkit.org/show_bug.cgi?id=258603

Reply:
So it seems like checking the document url's host and using `RegistrableDomain` would both be valid for this scenario. I changed to using `RegistrableDomain` just to make it shorter